lib/base_service.js

```diff
@@ -91,6 +120,18 @@ class BaseService {
             this._logger = new Logger(config.logging);
             this._setUpDNSCaching();
 
+            // Catch unhandled rejections & log them. This relies on bluebird.
+            if (!this._unhandledRejectionHandlerAdded) {
+                this._unhandledRejectionHandlerAdded = true;
+                process.on('unhandledRejection', this._logUnhandledException);
```

Comment:
There's still this boolean we discussed we don't need.

Reply:
Removed